spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/builder/Products.java

```diff
@@ -43,7 +43,8 @@ public static ProductDTO random() {
      *  a product DTO with a randomly generated SKU ID and name
      */
     public static ProductDTO randomEng() {
-        String id = StringUtil.random(8, StringUtil.CHARSET_NUMERIC);
+        // replace is needed because this id is used in a OID. BC does not allow leading zeros in a segment
+        String id = StringUtil.random(8, StringUtil.CHARSET_NUMERIC).replace('0', '1');
```

Comment:
Same here -- don't universally replace when you can just remove the bad option from the character set entirely.